### DISCUSSION TITLE
How would I go about making a new a chat tab that is separate from Global?

### DISCUSSION THREAD

**Discussion by Melodycloudu** [Earu/EasyChat#111] [Q&A] [answered]
is there a nice and neat way to make a new chat tab that allows for sending and recieving of messages that isnt apart of global?

**Comment by Earu** [comment]
Hi there ! I suggest following what I've done for the DM tab there https://github.com/Earu/EasyChat/blob/master/lua/easychat/modules/dm_tab.lua

EasyChat is a modular addon and as such any of its public functions should be available to your script if you put it under lua/easychat/modules/

**Reply by Melodycloudu** [reply]
is there a possiblity I could make it rank restricted with the ranks features included with easy chat? cause when looking through the wiki theres nothing about a way to use the ranks but yet theres ranks? and chat restrict

**Reply by Earu** [reply, marked as answer]
Well there are multiple ways you can do this, the admin tab (https://github.com/Earu/EasyChat/blob/master/lua/easychat/modules/admin_tab.lua#L217) might be closer to what you're looking for in that case.

Here's also another example of another script I made to add a tab: https://gist.github.com/Earu/b22ec3af9e8206bf16cbdcfc6391adcb
